aten/src/ATen/native/BatchLinearAlgebraKernel.cpp

```diff
@@ -253,6 +254,16 @@ void apply_lapack_eigh(const Tensor& values, const Tensor& vectors, const Tensor
   char jobz = compute_eigenvectors ? 'V' : 'N';
 
   auto n = vectors.size(-1);
+  
+  // Check that matrix size doesn't exceed INT_MAX
+  // LAPACK functions use 'int' for the leading dimension parameter, so
+  // matrices larger than INT_MAX cannot be processed due to overflow
+  TORCH_CHECK(n <= INT_MAX, 
+      "linalg.eigh: Matrix size (", n, ") exceeds the maximum value (", INT_MAX, 
+      ") supported by LAPACK. The LAPACK routines used by eigh internally ",
+      "require 32-bit integer parameters, which causes overflow for very large matrices. ",
+      "Please consider using a matrix of size at most ", INT_MAX, " or splitting your problem.");
+  
```

Comment:
Could you please revert the changes to this file? We try to keep kernels code as light wrappers around the library calls with all validation happening elsewhere.

The changes to `aten/src/ATen/native/BatchLinearAlgebra.cpp` are sufficient as the check error function is called in https://github.com/pytorch/pytorch/blob/6ab962e399c3689bdca9ce87fc5beb73864359b4/aten/src/ATen/native/BatchLinearAlgebra.cpp#L2843
```suggestion
```